### ISSUE TITLE
Inconsistent handling of `argv`

### ISSUE THREAD

**Issue by totpet** [SWI-Prolog/swipl-devel#1063]
I ran into a strange problem where the `argv` Prolog flag is different depending on when I'm checking it: `--` is left there when `current_prolog_flag` is called from a directive, but it's removed when a goal is being executed. Example:

`a.pl`:
```prolog
:- module(a, [main/0]).

:-
    current_prolog_flag(argv, Argv),
    writeln(1-Argv),
    argv_options(Argv, _, Options),
    writeln(1-Options).

main :-
    current_prolog_flag(argv, Argv),
    writeln(2-Argv),
    argv_options(Argv, _, Options),
    writeln(2-Options).

:- main.
```

```bash
$ swipl  -l a.pl -g main -g 'current_prolog_flag(argv, Argv), writeln(3-Argv)'  -- --a=b
1-[--,--a=b]
1-[]
2-[--,--a=b]
2-[]
2-[--a=b]
2-[a(b)]
3-[--a=b]
```

---

The [doc](https://www.swi-prolog.org/pldoc/doc_for?object=current_prolog_flag/2) says:
> Note that Prolog's argument processing stops at `--` or the first non-option argument

I can see this being interpreted both ways, I would prefer if `--` was always removed.

**Comment by JanWielemaker** [comment]
Thanks.   Should be fixed with f45ed53087df8893073c56c0b1b90e2482e782bf

Note that the normal way to invoke Prolog with scripts is

    swipl [prolog options] file1.pl file2.pl ... [script args]

And use `:- initialization(my_start_goal, main).` to set the start goal in one of the script files.

**Comment by totpet** [comment]
Thanks for the quick response!
